samples/kvs_gstreamer_sample.cpp

```diff
@@ -707,7 +710,10 @@ int gstreamer_live_source_init(int argc, char* argv[], CustomData *data, GstElem
             isOnRpi = false;
         }
         source = gst_element_factory_make("v4l2src", "source");
-        if (!source) {
+        if (source) {
+            LOG_DEBUG("Using v4l2src");
+        } else {
+            LOG_DEBUG("Using ksvideosrc");
             source = gst_element_factory_make("ksvideosrc", "source");
```

Comment:
Hmm I know this was always there but this looks Windows Specific [https://gstreamer.freedesktop.org/documentation/winks/index.html?gi-language=c]... so basically we're assuming if none of the other sources succeeded then it's Windows.  That's not good.  At the very least we should have a nul check here as well and log that it failed.  I understand in the  lines below we will pint error Not all elements could be created.  But I think it's better to explicitly log if source is still null that this element specifically couldn't be created.

Reply:
Added